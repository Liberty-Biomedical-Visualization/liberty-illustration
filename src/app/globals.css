@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --accent-50: 224 247 252;
  --accent-100: 178 224 240;
  --accent-200: 128 200 232;
  --accent-300: 77 176 215;
  --accent-400: 26 153 206;
  --accent-500: 3 152 206;
  --accent-600: 2 128 165;
  --accent-700: 2 109 136;
  --accent-800: 2 90 107;
  --accent-900: 2 61 77;
  --accent-950: 1 44 58;
  --background: 255 255 255;
  --foreground: 23 23 23;
}

body {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}
