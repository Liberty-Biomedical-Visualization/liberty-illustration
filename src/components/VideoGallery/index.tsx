import type { VideoGallery } from "@/lib/content";
import resolveClassNames from "@/lib/resolveClassNames";

import Video from "../Video";

export default function VideoGallery(props: Readonly<VideoGalleryProps>) {
  const { gallery } = props;

  const className = resolveClassNames(baseClassName, props.className);

  const videos = gallery.videos.map((videoData, index) => (
    <Video key={index} videoData={videoData} />
  ));

  return <div className={className}>{videos}</div>;
}

export interface VideoGalleryProps {
  className?: string;
  gallery: Readonly<VideoGallery>;
}

const baseClassName = "flex flex-col gap-4";
