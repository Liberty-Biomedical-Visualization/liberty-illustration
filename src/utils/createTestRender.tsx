import { render } from "@testing-library/react";
import type { JSX, JSXElementConstructor } from "react";

/**
 * Returns a function that allows rendering Component with partial props while
 * falling back to defaultProps for testing purposes.
 */
export default function createTestRender<C extends JSXElementConstructor<P>, P>(
  Component: JSXElementConstructor<P> & C,
  defaultProps: JSX.LibraryManagedAttributes<C, P>,
) {
  return function (props: Partial<P> = {}) {
    const resolvedProps: JSX.IntrinsicAttributes &
      JSX.LibraryManagedAttributes<C, P> = { ...defaultProps, ...props };
    return render(<Component {...resolvedProps} />);
  };
}
