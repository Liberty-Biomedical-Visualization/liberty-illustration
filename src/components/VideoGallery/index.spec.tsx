import { describe, expect, it } from "@jest/globals";
import { screen } from "@testing-library/react";

import {
  completeVideoGallery,
  minimalVideoGallery,
} from "@/lib/content/test-video-gallery-data";
import createTestRender from "@/lib/createTestRender";

import VideoGallery from ".";

describe("VideoGallery", () => {
  it("should display videos when the gallery has video data", () => {
    renderVideoGallery({ gallery: completeVideoGallery });

    for (const videoData of completeVideoGallery.videos) {
      const video = screen.getByRole("figure", { name: videoData.title });
      expect(video).toBeVisible();
    }
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderVideoGallery({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderVideoGallery();
    expect(result.container).toMatchSnapshot();
  });
});

const renderVideoGallery = createTestRender(VideoGallery, {
  gallery: minimalVideoGallery,
});
