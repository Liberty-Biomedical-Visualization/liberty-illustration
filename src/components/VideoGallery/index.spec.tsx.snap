// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGallery should match the snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  />
</div>
`;
